x/msg_authorization/internal/types/msgs.go

```diff
@@ -81,45 +82,46 @@ func (msg MsgRevokeAuthorization) GetSignBytes() []byte {
 	return sdk.MustSortJSON(bz)
 }
 
-func (msg MsgRevokeAuthorization) ValidateBasic() sdk.Error {
+func (msg MsgRevokeAuthorization) ValidateBasic() error {
 	if msg.Granter.Empty() {
-		return sdk.ErrInvalidAddress(msg.Granter.String())
+		return sdkerrors.Wrap(ErrInvalidGranter, "missing granter address")
 	}
 	if msg.Grantee.Empty() {
-		return sdk.ErrInvalidAddress(msg.Grantee.String())
+		return sdkerrors.Wrap(ErrInvalidGranter, "missing grantee address")
 	}
 	return nil
 }
 
-// MsgExecDelegated attempts to execute the provided messages using
-// capabilities granted to the grantee. Each message should have only
-// one signer corresponding to the granter of the capability.
-type MsgExecDelegated struct {
+// MsgExecAuthorized attempts to execute the provided messages using
+// authorizations granted to the grantee. Each message should have only
+// one signer corresponding to the granter of the authorization.
+type MsgExecAuthorized struct {
 	Grantee sdk.AccAddress `json:"grantee"`
-	Msgs    []sdk.Msg      `json:"msg"`
+	Msgs    []sdk.Msg      `json:"msgs"`
 }
 
-func NewMsgExecDelegated(grantee sdk.AccAddress, msgs []sdk.Msg) MsgExecDelegated {
-	return MsgExecDelegated{
+func NewMsgExecAuthorized(grantee sdk.AccAddress, msg []sdk.Msg) MsgExecAuthorized {
+	return MsgExecAuthorized{
 		Grantee: grantee,
-		Msgs:    msgs,
+		Msgs:    msg,
 	}
 }
-func (msg MsgExecDelegated) Route() string { return RouterKey }
-func (msg MsgExecDelegated) Type() string  { return "exec_delegated" }
 
-func (msg MsgExecDelegated) GetSigners() []sdk.AccAddress {
+func (msg MsgExecAuthorized) Route() string { return RouterKey }
+func (msg MsgExecAuthorized) Type() string  { return "exec_delegated" }
+
+func (msg MsgExecAuthorized) GetSigners() []sdk.AccAddress {
 	return []sdk.AccAddress{msg.Grantee}
 }
 
-func (msg MsgExecDelegated) GetSignBytes() []byte {
+func (msg MsgExecAuthorized) GetSignBytes() []byte {
 	bz := ModuleCdc.MustMarshalJSON(msg)
```

Comment:
We're receiving a panic when we try to create a send-as tx(from gaia) as follows,` panic: Unregistered interface types.Msg`
Couldn't figure out the issue, but this is where the panic seems to be coming from https://github.com/tendermint/go-amino/blob/59d50ef176f686dc2c0c150e85a9381a78b9978a/amino.go#L369

Reply:
Oh the module codec doesn't actually have the full app codec that has all the `sdk.Msg`s registered.... That's not good. I'll try to think of a work-around, but I have a hunch it may be best to start moving to protobuf and not support amino here because the signing logic will be different there anyway. Wonder what @alexanderbez thinks.

Reply:
I agree with the idea. In the proto PRs we've been doing, we've been leaving `GetSignBytes` alone. That is, it still calls `ModuleCdc.MustMarshalJSON(msg)`. Eventually, this will be removed entirely or modified at least.

Reply:
Okay, so sounds like we should start moving this to protobuf right away and not worry about supporting amino. However, we will need to figure out the protobuf `Msg` approach and how that works from the CLI, etc to get this finished.